src/actinia_core/core/common/process_chain.py

```diff
@@ -478,169 +520,232 @@ def _create_module_process(self, module_descr):
-                    # Generate the temporary file path and store it in the dict
-                    if file_id not in self.temporary_pc_files:
-                        self.temporary_pc_files[file_id] = self.generate_temp_file_path(
-                        )
-                    # Store the file path in the output description for export
-                    param = "%s=%s" % (param, self.temporary_pc_files[file_id])
-                    # Add the temp file path and the new file name with suffix to the
-                    # output dict
-                    if "export" in output:
-                        output["tmp_file"] = self.temporary_pc_files[file_id]
-                        output["file_name"] = "%s.%s" % (
-                            file_id, output["export"]["format"].lower())
-                else:
-                    param = "%s=%s" % (param, value)
-                params.append(param)
-
-                # save the output dict in a resource export list
+            # Search for file identifiers and generate the temporary file path
+            if "$file" in value and "::" in value:
+                file_id = value.split("::")[1]
+                # Generate the temporary file path and store it in the dict
+                if file_id not in self.temporary_pc_files:
+                    self.temporary_pc_files[file_id] = \
+                            self.generate_temp_file_path()
+                # Store the file path in the output description for export
+                param = "%s=%s" % (param, self.temporary_pc_files[file_id])
+                # Add the temp file path and the new file name with suffix to the,
+                # output dict
                 if "export" in output:
-                    self.resource_export_list.append(output)
-
-        if "flags" in module_descr:
-            if "flags" in module_descr:
-                params.append("-" + str(module_descr["flags"]))
-
-        if "overwrite" in module_descr and module_descr["overwrite"] is True:
-            params.append("--o")
-
-        if "superquiet" in module_descr and module_descr["superquiet"] is True:
-            params.append("--qq")
-
-        if "verbose" in module_descr and module_descr["verbose"] is True:
-            params.append("--v")
+                    output["tmp_file"] = self.temporary_pc_files[file_id]
+                    output["file_name"] = "%s.%s" % (
+                            file_id, output["export"]["format"].lower())
+            else:
+                param = "%s=%s" % (param, value)
+            params.append(param)
+            # save the output dict in a resource export list
+            if "export" in output:
+                self.resource_export_list.append(output)
 
-        if ("interface-description" in module_descr
-                and module_descr["interface-description"] is True):
-            params.append("--interface-description")
+    def _check_stdout(self, module_descr, id):
```

Comment:
Perspectively, maybe we could move all `_check...`, `_create...` etc. methods together?